#!/usr/bin/env python
PACKAGE='barrier_selection'
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#       Name                    Type      Reconfiguration level
#       Description
#       Default  Min  Max
gen.add("barrier_box_x_adjust",    double_t,    0, "Barrier Box X", 0.0,  -5.0, 5.0)
gen.add("barrier_box_y_adjust",    double_t,    0, "Barrier Box Y", 0.0,  -5.0, 5.0)
gen.add("barrier_box_z_adjust",    double_t,    0, "Barrier Box Z", 0.0,  -5.0, 5.0)
gen.add("barrier_box_width_adjust" ,    double_t,    0, "Barrier Box Width" ,  0.0, -5.0, 5.0)
gen.add("barrier_box_height_adjust",    double_t,    0, "Barrier Box Height",  0.0, -5.0, 5.0)
gen.add("barrier_box_length_adjust",    double_t,    0, "Barrier Box Length",  0.0, -5.0, 5.0)
gen.add("barrier_stopline_matcher_radius",    double_t,    5, "Barrier-Stopline Matcher Radius",  5.0, 0, 40.0)

exit(gen.generate(PACKAGE, "barrier_selection", "BarrierSelection"))